src/project/internal/notationproject.cpp

```diff
@@ -128,18 +128,36 @@ void NotationProject::setupProject()
     m_engravingProject = EngravingProject::create();
     m_engravingProject->setFileInfoProvider(std::make_shared<ProjectFileInfoProvider>(this));
 
+    auto listenNotationChanges = [this](INotationPtr notation) {
+        INotationWeakPtr weakPtr = notation;
+
+        notation->undoStack()->changesChannel().onReceive(this, [this, weakPtr](const ScoreChangesRange&) {
```

Comment:
Since there is essentially only one undo stack for the whole project (I mean, there is one _underlying_ UndoStack, namely in the `MasterScore`), it should be enough to listen only for changes from the MasterScore / MasterNotation. The notifications from the excerpts are only duplicates.

Reply:
Thanks! Actually, this code was only working because it was inside the loop. changesChannel() was not available for MasterNotation in setupProject() (because m_score had not yet been initialized), but became available after receiving some changes in the excerpt list. Now this code should be more robust